### ISSUE TITLE
Cant recognize Code128 barcode

### ISSUE THREAD

**Issue by PerlAmutor** [zxing-cpp/zxing-cpp#779]
I try use imagemagic before scan, but it doesn't help

` convert -density 150 -resize 1000% -threshold 55% 000001.png out.png`

![000001_orig](https://github.com/zxing-cpp/zxing-cpp/assets/87958986/22e88a3f-cf26-4ea9-abf5-19d545a5f6b5)
![000001](https://github.com/zxing-cpp/zxing-cpp/assets/87958986/8a4ea678-3dbc-4a88-9890-3cecff6e0eb0)

After use 'convert' zbar read barcode without problem, but not zxing-cpp.

**Comment by axxel** [comment]
If I apply your 'convert' line above and use current HEAD's `ZXingReader` it can also detect this sample. Did you try with an outdated version?

**Comment by PerlAmutor** [comment]
I build ZXingReader from sources 26.04.2024 (cloning master).

**Comment by axxel** [comment]
Here is a 5x (instead of 10x) scaled version (both work for me):
![#779 scaled](https://github.com/zxing-cpp/zxing-cpp/assets/1038346/5d588def-dc34-423c-9278-22b85aeeab48)
I used ImageMagick 6.9.11-60 Q16 x86_64 2021-01-25

**Comment by PerlAmutor** [comment]
Yes, it both working for 500% and 1000% but not working for fullpage (A4) scan. Looks like I don't try scan image after crop full picture before post here. As I understand ZXing-cpp doesnt have function for find POI with potential barcodes and extract it to separate small images before try to validate?

**Comment by axxel** [comment]
> As I understand ZXing-cpp doesnt have function for find POI with potential barcodes and extract it to separate small images before try to validate?

No it does not but it should work in this case anyway. Can you provide the full page (converted to png/jpg) that does not work with `zxing-cpp` but does so with zbar? I need to be able to reproduce the actual issue, otherwise I can't help you.

**Comment by PerlAmutor** [comment]
> Can you provide the full page

Sure. I just change Canvas size and now it's not detectable.
![000001_A4](https://github.com/zxing-cpp/zxing-cpp/assets/87958986/108b6c61-37de-4560-8a32-025079269643)

**Comment by axxel** [comment]
But here, you started with the sample that did not work in the first place. Adding whitespace around that does of course not make it work. Are you sure this sample works with zbar? That would sound like a contradiction to your earlier statements.

**Comment by PerlAmutor** [comment]
> Are you sure this sample works with zbar?
Yes, it worked with ZBar after ImageMagick command from first post

![image](https://github.com/zxing-cpp/zxing-cpp/assets/87958986/e5afc532-60e6-41bb-93e9-90fc93fba658)

**Comment by axxel** [comment]
You are right, simply adding whitespace breaks the detection in this case. This is due to [this line](https://github.com/zxing-cpp/zxing-cpp/blob/38da404a43d19dfe9f85fd19d57869a8fa8ae712/core/src/oned/ODReader.cpp#L95). It is a means to 'balance' runtime performance vs. detection performance. If you simply replace that line with `int rowStep = 1;` it should work. Maybe that is an option for your use-case?
